src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchive.Create.Unix.cs

```diff
@@ -1,16 +1,28 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics;
+
+using static System.IO.Compression.ZipArchiveEntryConstants;
+
 namespace System.IO.Compression
 {
     public static partial class ZipFileExtensions
     {
         static partial void SetExternalAttributes(FileStream fs, ZipArchiveEntry entry)
         {
+            Debug.Assert(!OperatingSystem.IsWindows());
+
+            // SetExternalAttributes is only used to overwrite the default values when reading on the unix systems
+            // This assert ensures that nothing else sets values different to the default before
+            // overwriting it with the data read from the files
+            Debug.Assert(entry.ExternalAttributes == ((uint)DefaultFileEntryPermissions << 16)
+                         || entry.ExternalAttributes == ((uint)DefaultDirectoryEntryPermissions << 16));
```

Comment:
This is only ever called for files (i.e. it takes a `FileStream`), so we shouldn't need to check for `DefaultDirectoryEntryPermissions`.

Reply:
Removed the `DefaultDirectoryEntryPermissions` check. I wanted to first change this method to also work with the directories but then decided not to do it in this PR.